intel_extension_for_transformers/llm/runtime/graph/models/chatglm/chatglm1.cpp

```diff
@@ -0,0 +1,487 @@
+    // printf("logits_out: ");
+    // for (int i = 0; i < 20; i++) {
+    //   printf("%f, ", logits_out[i]);
+    // }
+    // printf("\n");
+  }
+
+  // extract embeddings
+  if (!lctx.embedding.empty()) {
+    auto& embedding_out = lctx.embedding;
+
+    embedding_out.resize(n_embd);
+    memcpy(embedding_out.data(), (float*)ne_get_data(embeddings) + (n_embd * (N - 1)), sizeof(float) * n_embd);
+  }
+
+  if (mem_per_token == 0) {
+    mem_per_token = ne_used_mem(ctx0) / N;
+  }
+
+  ne_free(ctx0);
+
+  // measure the performance only for the single-token evals
+  int64_t time_interval = ne_time_us() - t_start_us;
+  if (N == 1) {
+    lctx.t_eval_us += time_interval;
+    lctx.n_eval++;
+  } else if (N > 1) {
+    lctx.t_p_eval_us += time_interval;
+    lctx.n_p_eval += N;
+  }
+  lctx.eval_times.push_back(time_interval);
+
+  return true;
+}
+
+int model_eval(struct model_context* ctx, const model_token* tokens, int n_tokens, int n_past, int n_threads) {
+  if (!chatglm_model_eval_internal(*ctx, tokens, n_tokens, n_past, n_threads)) {
+    fprintf(stderr, "%s: failed to eval\n", __func__);
+    return 1;
+  }
+
+  // get a more accurate load time, upon first eval
+  // TODO: fix this
+  if (!ctx->has_evaluated_once) {
+    ctx->t_load_us = ne_time_us() - ctx->t_start_us;
+    ctx->has_evaluated_once = true;
+  }
+
+  return 0;
+}
+
+// TODO: not great allocating this every time
+// std::vector<model_token> model_tokenize(struct model_context* ctx, const std::string& text, bool add_bos) {
+//   // initialize to prompt numer of chars, since n_tokens <= n_prompt_chars
+//   std::vector<model_token> res(text.size() + (int)add_bos);
+//   const int n = model_tokenize(ctx, text.c_str(), res.data(), res.size(), add_bos);
+//   assert(n >= 0);
+//   res.resize(n);
+
+//   return res;
+// }
+
+// struct model_context* model_init_from_gpt_params(const gpt_params& params) {
+//   auto lparams = model_context_default_params();
+
+//   lparams.name = params.name;
+//   lparams.n_ctx = params.n_ctx;
+//   lparams.n_gpu_layers = params.n_gpu_layers;
+//   lparams.seed = params.seed;
+//   lparams.f16_kv = params.memory_f16;
+//   lparams.use_mmap = params.use_mmap;
+//   lparams.use_mlock = params.use_mlock;
+//   lparams.logits_all = params.perplexity;
+//   lparams.embedding = params.embedding;
+
+//   model_context* lctx = model_init_from_file(params.model.c_str(), lparams);
+
+//   if (lctx == NULL) {
+//     fprintf(stderr, "%s: error: failed to load model '%s'\n", __func__, params.model.c_str());
+//     return NULL;
+//   }
+
+//   if (!params.lora_adapter.empty()) {
+//     int err = model_apply_lora_from_file(lctx, params.lora_adapter.c_str(),
+//                                          params.lora_base.empty() ? NULL : params.lora_base.c_str(), params.n_threads);
+//     if (err != 0) {
+//       fprintf(stderr, "%s: error: failed to apply lora adapter\n", __func__);
+//       return NULL;
+//     }
+//   }
+
+//   return lctx;
+// }
```

Comment:
maybe need to clean some useless annotations in this file

Reply:
Yes. I will do cleancode in these files of chatglm-1.

This PR is for the chatglm-2 enabling.